ccdproc/core.py

```diff
@@ -1558,6 +1558,134 @@ def cosmicray_median(ccd, error_image=None, thresh=5, mbox=11, gbox=0,
+    unaffected by bad pixels provided the median size is larger that twice the
+    size of a bad region. Thus, if 3 pixel wide bad columns are present then the
+    column median box size should be at least 7 pixels. The median box can be a
+    single pixel wide along one dimension if needed. This may be appropriate for
+    spectroscopic long slit data.
+
+    The median subtracted image is then divided into blocks of size nclsig by
+    nlsig. In each block the pixel values are sorted and the pixels nearest the
+    30.9 and 69.1 percentile points are found; this would be the one sigma
+    points in a Gaussian noise distribution. The difference between the two
+    count levels divided by two is then the local sigma estimate. This algorithm
+    is used to avoid contamination by the bad pixel values. The block size must
+    be at least 10 pixels in each dimension to provide sufficient pixels for a
+    good estimate of the percentile sigma. The sigma uncertainty estimate of
+    each pixel in the image is then the sigma from the nearest block.
+
+    The deviant pixels are found by comparing the median subtracted residual to
+    a specified sigma threshold factor times the local sigma above and below
+    zero (the lsigma and hsigma parameters). This is done for individual pixels
+    and then for column sums of pixels (excluding previously flagged bad pixels)
+    from two to the number of lines in the image. The sigma of the sums is
+    scaled by the square root of the number of pixels summed so that
+    statistically low or high column regions may be detected even though
+    individual pixels may not be statistically deviant. For the purpose of this
+    task one would normally select large sigma threshold factors such as six or
+    greater to detect only true bad pixels and not the extremes of the noise
+    distribution.
+
+    As a final step each column is examined to see if there are small segments
+    of unflagged pixels between bad pixels. If the length of a segment is less
+    than that given by the ngood parameter all the pixels in the segment are
+    also marked as bad.
+
+
+    Returns
+    -------
+    mask : `numpy.ndarray`
+        A boolean ndarray with the bad pixel positions identified.
+    '''
+    if ratio.data.ndim !=2:
+        return None
+    mask = ~np.isfinite(ratio.data)
+    medsub = ratio.data - ndimage.filters.median_filter(ratio.data,
+                                                        size=(nlsig, ncsig))
+    nl, nc = ratio.data.shape
+    nbl = int(np.ceil(ratio.data.shape[0] / nlsig))
+    nbc = int(np.ceil(ratio.data.shape[1] / ncsig))
+    for i in range(nbl):
+        for j in range(nbc):
+            l1 = i*nlsig
+            l2 = min((i+1)*nlsig, nl)
+            c1 = j*ncsig
+            c2 = min((j+1)*ncsig, nc)
+            block = medsub[l1:l2,c1:c2]
+            high = np.percentile(block.ravel(), 69.1)
```

Comment:
What block-sizes did you have for the sigma-calculation? 

I ask because the byblocks variant scales with t ~ 1/A (A = block_x * block_y) while the filter variant scales with t ~ A.